src/components/ResetPassword.js

```diff
@@ -0,0 +1,54 @@
+// ForgotPassword.js
+import React, { useState } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { resetPassword } from '../apis/authApi'; // Make sure to import your API call function
+
+const ResetPassword = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const token = searchParams.get('token');
+
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [error, setError] = useState('');
+
+  const handleResetPassword = async () => {
+    try {
+      if (password !== confirmPassword) {
+        setError('Passwords do not match');
+        return;
+      }
+
+      await resetPassword({ token, password });
+
+      // Redirect to sign in page on success
+      navigate.push('/signin');
```

Comment:
`navigate` doesn't have a `.push` function:

[https://github.com/remix-run/react-router/blob/main/docs/upgrading/v5.md#use-usenavigate-instead-of-usehistory](https://github.com/remix-run/react-router/blob/main/docs/upgrading/v5.md#use-usenavigate-instead-of-usehistory)

^ The example there shows that instead of `history.push` , now in React-Router v6 we can pass the url path we want to navigate to directly into `navigate` as a function call: `navigate("/signIn")`

Reply:
You are right , sorry, commit bb3122006976aea093071aa9dc2bbbd7abc4278d